hudi-common/src/main/java/org/apache/hudi/keygen/constant/KeyGeneratorType.java

```diff
@@ -150,4 +150,12 @@ public static String getKeyGeneratorClassName(Map<String, String> config) {
     }
     return null;
   }
+
+  public KeyGeneratorType getAvroImplementation() {
+    if (name().endsWith("AVRO")) {
+      return this;
+    }
+    String avroName = name() + "_AVRO";
+    return KeyGeneratorType.valueOf(avroName);
```

Comment:
There are certain key generator types that do not have "AVRO" counterpart, e.g., `KeyGeneratorType.HOODIE_TABLE_METADATA` (which is the only important one to take care of).  Should that be accounted for since MDT read will use `HoodieAvroReaderContext`?

Reply:
I'm thinking that this should be moved into the avro reader context class and we can inspect whether the class is on the path before attempting this. How does that sound?

Reply:
This is no longer relevant.  Yet we should still see if we can get rid of `HoodieTableMetadataKeyGenerator` to simplify the logic.